### PR TITLE
fix(reports): make name unique between alerts and reports

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Uniqueness was being imposed on Alerts/Report names, preventing user's from creating an Alert name "X" and a Report with the same name "X". This allows it while still enforcing name uniqueness on Alerts and Reports separately

Also guarantees that a database reference is not sent when creating a report

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/migrations/versions/c878781977c6_alert_reports_shared_uniqueness.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            op.drop_constraint(
+                "report_schedule_name_key", "report_schedule", type_="unique"
+            )
+        op.create_unique_constraint(
+            "uq_report_schedule_name_type", "report_schedule", ["name", "type"]
+        )
+
+    else:
+        with op.batch_alter_table(
+            "report_schedule", copy_from=report_schedule
```
Comment (dpgaspar) [inline superset/migrations/versions/c878781977c6_alert_reports_shared_uniqueness.py:96]: This was required to remove `UNIQUE (name)` that is not an actual constraint from SQLite

**Review by nytai — APPROVED** [review]
LGTM. SQLite constraints are such a pain to deal with, we should consider dropping official support for it soon.

File: superset/migrations/versions/c878781977c6_alert_reports_shared_uniqueness.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        ) as batch_op:
+            batch_op.create_unique_constraint(
+                "uq_report_schedule_name_type", ["name", "type"]
+            )
```
Comment (villebro) [inline superset/migrations/versions/c878781977c6_alert_reports_shared_uniqueness.py:100]: This took a loong while for me to wrap my head around 😆 A small comment here could be really helpful for the next person trying to understand what's going on.

**Review by villebro — APPROVED** [review]
LGTM!

**Pull request merged**